parser/src/lib.rs

```diff
@@ -93,61 +93,55 @@ impl<'a> Parser<'a> {
         }))
     }
 
-    fn match_token(&mut self, expected: TokenType) -> bool {
-        if let Some(actual) = self.tokens.get(self.current) {
-            if actual.token_type == expected {
-                self.current += 1;
-                true
-            } else {
-                false
-            }
-        } else {
-            false
+    fn exists_token(&mut self, expected: TokenType) -> bool {
+        let token = self.peek_token();
+        if token.token_type == expected {
+            self.current += 1;
```

Comment:
Not a fan of it being named `exists` and actually mutating the `current` token cursor.